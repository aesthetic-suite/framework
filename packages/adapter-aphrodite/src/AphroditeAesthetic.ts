import Aesthetic, { AestheticOptions, ClassName, Ruleset, Sheet, SheetMap } from 'aesthetic';
// @ts-ignore flushToStyleTag is not typed
import { StyleSheet as Aphrodite, Extension, flushToStyleTag } from 'aphrodite';
import { NativeBlock, ParsedBlock } from './types';

export default class AphroditeAesthetic<Theme extends object> extends Aesthetic<
  Theme,
  NativeBlock,
  ParsedBlock
> {
  aphrodite: {
    StyleSheet: typeof Aphrodite;
    css(...styles: ParsedBlock[]): ClassName;
  };

  fontFaces: { [fontFamily: string]: NativeBlock[] } = {};

  keyframes: { [animationName: string]: NativeBlock } = {};

  constructor(extensions: Extension[] = [], options: Partial<AestheticOptions> = {}) {
    super(options);

    this.aphrodite = Aphrodite.extend([
      ...extensions,
      { selectorHandler: this.handleHierarchySelector },
      { selectorHandler: this.handleGlobalSelector },
    ]);

    this.syntax
      .on('attribute', this.handleNested)
      .on('css', this.handleCss)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframe', this.handleKeyframe)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested);
  }

  flushStyles() {
    flushToStyleTag();
  }

  isParsedBlock(block: NativeBlock | ParsedBlock): block is ParsedBlock {
    // eslint-disable-next-line no-underscore-dangle
    return Boolean(block && block._name && block._definition);
  }

  protected processStyleSheet(styleSheet: SheetMap<NativeBlock>): SheetMap<ParsedBlock> {
    return this.aphrodite.StyleSheet.create(styleSheet) as SheetMap<ParsedBlock>;
  }

  protected transformToClassName(styles: ParsedBlock[]): ClassName {
    return this.aphrodite.css(...styles);
  }

  private handleCss = (css: string) => {
    this.getStyleSheetManager().injectStatements(css);
  };

  // https://github.com/Khan/aphrodite#font-faces
  private handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: Ruleset<NativeBlock>[],
    fontFamily: string,
  ) => {
    this.fontFaces[fontFamily] = fontFaces.map(face => face.toObject());
  };

  private handleGlobal = (
    sheet: Sheet<NativeBlock>,
    selector: string,
    ruleset: Ruleset<NativeBlock>,
  ) => {
    const current: Ruleset<NativeBlock> = sheet.ruleSets.globals || sheet.createRuleset('globals');

    current.addNested(`*${selector}`, ruleset);

    sheet.addRuleset(current);
  };

  private handleGlobalSelector(
    selector: string,
    baseSelector: string,
    callback: (selector: string) => string,
  ): string | null {
    if (selector.charAt(0) !== '*') {
      return null;
    }

    return callback(selector.slice(1));
  }

  private handleHierarchySelector(
    selector: string,
    baseSelector: string,
    callback: (selector: string) => string,
  ): string | null {
    if (selector.charAt(0) === '>') {
      return callback(`${baseSelector} ${selector}`);
    }

    if (selector.charAt(0) === '[') {
      return callback(`${baseSelector}${selector}`);
    }

    return null;
  }

  // https://github.com/Khan/aphrodite#animations
  private handleKeyframe = (
    sheet: Sheet<NativeBlock>,
    keyframe: Ruleset<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = keyframe.toObject();
  };

  // https://github.com/Khan/aphrodite#api
  private handleMedia = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // https://github.com/Khan/aphrodite#api
  private handleNested = (
    ruleset: Ruleset<NativeBlock>,
    selector: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(selector, value);
  };

  // https://github.com/Khan/aphrodite#api
  private handleProperty = (ruleset: Ruleset<NativeBlock>, name: keyof NativeBlock, value: any) => {
    if (name === 'animationName') {
      ruleset.addProperty(name, this.syntax.injectKeyframes(value, this.keyframes));
    } else if (name === 'fontFamily') {
      // Font faces could potentially convert recursively because font faces
      // have a `familyName`, and we parse on `familyName`. Luckily the ruleset
      // selector is the family name, and we can determine that this is being
      // called from from `convertFontFaces`.
      if (value.includes(ruleset.selector)) {
        ruleset.addProperty(name, value);
      } else {
        ruleset.addProperty(name, this.syntax.injectFontFaces(value, this.fontFaces));
      }
    } else {
      ruleset.addProperty(name, value);
    }
  };
}
