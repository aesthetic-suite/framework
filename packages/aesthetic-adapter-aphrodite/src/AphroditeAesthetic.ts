/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import Aesthetic, { AestheticOptions, ClassName, Ruleset, Sheet, StyleSheetMap } from 'aesthetic';
import { StyleSheet as Aphrodite, Extension } from 'aphrodite';
import { NativeBlock, ParsedBlock } from './types';

export default class AphroditeAesthetic<Theme> extends Aesthetic<Theme, NativeBlock, ParsedBlock> {
  aphrodite: {
    css(...styles: ParsedBlock[]): ClassName;
    StyleSheet: typeof Aphrodite;
  };

  fontFaces: { [fontFamily: string]: NativeBlock[] } = {};

  keyframes: { [animationName: string]: NativeBlock } = {};

  constructor(extensions: Extension[] = [], options: Partial<AestheticOptions> = {}) {
    super(options);

    this.aphrodite = Aphrodite.extend([
      ...extensions,
      { selectorHandler: this.handleHierarchySelector },
      { selectorHandler: this.handleGlobalSelector },
    ]);

    this.syntax
      .on('attribute', this.handleNested)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframe', this.handleKeyframe)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested);
  }

  protected processStyleSheet(styleSheet: object): StyleSheetMap<ParsedBlock> {
    return this.aphrodite.StyleSheet.create(styleSheet) as StyleSheetMap<ParsedBlock>;
  }

  protected transformToClassName(styles: (NativeBlock | ParsedBlock)[]): ClassName {
    const legitStyles: ParsedBlock[] = [];
    const tempStylesheet: { [key: string]: NativeBlock } = {};
    let counter = 0;

    styles.forEach(style => {
      // eslint-disable-next-line no-underscore-dangle
      if (style._name && style._definition) {
        legitStyles.push(style as ParsedBlock);
      } else {
        tempStylesheet[`inline-${counter}`] = style;
        counter += 1;
      }
    });

    if (counter > 0) {
      legitStyles.push(...Object.values(this.processStyleSheet(tempStylesheet)));
    }

    return this.aphrodite.css(...legitStyles);
  }

  // https://github.com/Khan/aphrodite#font-faces
  private handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: Ruleset<NativeBlock>[],
    fontFamily: string,
  ) => {
    this.fontFaces[fontFamily] = fontFaces.map(face => face.toObject());
  };

  private handleGlobal = (
    sheet: Sheet<NativeBlock>,
    selector: string,
    ruleset: Ruleset<NativeBlock>,
  ) => {
    const current: Ruleset<NativeBlock> = sheet.ruleSets.globals || sheet.createRuleset('globals');

    current.addNested(`*${selector}`, ruleset);

    sheet.addRuleset(current);
  };

  private handleGlobalSelector(
    selector: string,
    baseSelector: string,
    callback: (selector: string) => string,
  ): string | null {
    if (selector.charAt(0) !== '*') {
      return null;
    }

    return callback(selector.slice(1));
  }

  private handleHierarchySelector(
    selector: string,
    baseSelector: string,
    callback: (selector: string) => string,
  ): string | null {
    if (selector.charAt(0) === '>') {
      return callback(`${baseSelector} ${selector}`);
    }

    if (selector.charAt(0) === '[') {
      return callback(`${baseSelector}${selector}`);
    }

    return null;
  }

  // https://github.com/Khan/aphrodite#animations
  private handleKeyframe = (
    sheet: Sheet<NativeBlock>,
    keyframe: Ruleset<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = keyframe.toObject();
  };

  // https://github.com/Khan/aphrodite#api
  private handleMedia = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // https://github.com/Khan/aphrodite#api
  private handleNested = (
    ruleset: Ruleset<NativeBlock>,
    selector: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(selector, value);
  };

  // https://github.com/Khan/aphrodite#api
  private handleProperty = (ruleset: Ruleset<NativeBlock>, name: keyof NativeBlock, value: any) => {
    if (name === 'animationName') {
      ruleset.addProperty(name, this.syntax.injectKeyframes(value, this.keyframes));
    } else if (name === 'fontFamily') {
      ruleset.addProperty(name, this.syntax.injectFontFaces(value, this.fontFaces));
    } else {
      ruleset.addProperty(name, value);
    }
  };
}
