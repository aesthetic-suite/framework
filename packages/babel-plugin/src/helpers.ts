import { types as t } from '@babel/core';
import { createDebugger } from '@boost/debug';

export const debug = createDebugger('aesthetic:babel');

export function isFunction(node: t.Node) {
  return t.isFunctionExpression(node) || t.isArrowFunctionExpression(node);
}
