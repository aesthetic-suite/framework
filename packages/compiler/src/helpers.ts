import { ColorShade } from '@aesthetic/system';

export function formatShade(value: number | string): ColorShade {
	return String(value).padStart(2, '0').slice(0, 2) as ColorShade;
}

export function formatUnit(value: number): string {
	return value.toFixed(2).replace('.00', '');
}
