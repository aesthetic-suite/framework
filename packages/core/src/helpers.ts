import { Theme } from '@aesthetic/system';
import { objectLoop } from '@aesthetic/utils';
import { SheetParams } from './types';

export function createCacheKey(params: SheetParams, type: string): string {
  let key = type;

  // Since all other values are scalars, we can just join the values.
  // This is 3x faster than JSON.stringify(), and 1.5x faster than Object.values()!
  objectLoop(params, (value) => {
    key += value;
  });

  return key;
}

export function createDefaultParams(theme: Theme<{}>, params: SheetParams): Required<SheetParams> {
  return {
    contrast: theme.contrast,
    deterministic: false,
    direction: 'ltr',
    scheme: theme.scheme,
    theme: theme.name,
    unit: 'px',
    vendor: false,
    ...params,
  };
}
