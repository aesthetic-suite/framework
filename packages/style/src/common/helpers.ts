/* eslint-disable no-magic-numbers */

import { Value, MaybeValue, CSS, Sheet } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';
import { IMPORT_RULE, STYLE_RULE } from './constants';

export function insertRule(sheet: Sheet, rule: CSS, index?: number): number {
  try {
    return sheet.insertRule(rule, index ?? sheet.cssRules.length);
  } catch {
    // Vendor prefixed properties, pseudos, etc, that are inserted
    // into different vendors will trigger a failure. For example,
    // `-moz` or `-ms` being inserted into WebKit.
    // There's no easy way around this, so let's just ignore the
    // error so that subsequent styles are inserted.
    // istanbul ignore next
    return -1;
  }
}

export function insertAtRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // At-rules must be inserted before normal style rules.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type === STYLE_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function insertImportRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // Import rules must be inserted at the top of the style sheet,
  // but we also want to persist the existing order.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type !== IMPORT_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import';
}

export function isMediaRule(value: string): boolean {
  return value.slice(0, 6) === '@media';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

export function isSupportsRule(value: string): boolean {
  return value.slice(0, 9) === '@supports';
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      // eslint-disable-next-line no-console
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}

export function joinQueries(prev: string | undefined, next: string): string {
  if (prev) {
    return `${prev} and ${next}`;
  }

  return next;
}
