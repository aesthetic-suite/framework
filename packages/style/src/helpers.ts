/* eslint-disable no-magic-numbers */

import { Value, MaybeValue } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import' && value[7] === ' ';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}
