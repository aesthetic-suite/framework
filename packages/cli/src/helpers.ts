import { CONFIG_FOLDER } from '@aesthetic/compiler';
import { Path } from '@boost/common';
import { CWD } from './constants';

export function getConfigFolderDir(name: string): Path {
  return new Path(CWD, CONFIG_FOLDER, name);
}

export function validateSystemName(value: string) {
  if (value && !value.match(/^[-a-z0-9]+$/giu)) {
    throw new Error('Name may only contain numbers, letters, and dashes.');
  }
}
