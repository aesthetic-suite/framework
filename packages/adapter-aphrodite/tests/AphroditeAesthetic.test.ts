/* eslint-disable jest/expect-expect */

import { StyleSheetTestUtils } from 'aphrodite';
import AphroditeAesthetic from '../src/AphroditeAesthetic';
import {
  FONT_ROBOTO_FLAT_SRC,
  FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
  KEYFRAME_FADE,
  SYNTAX_UNIFIED_LOCAL_FULL,
  SYNTAX_ATTRIBUTE,
  SYNTAX_DESCENDANT,
  SYNTAX_FONT_FACE,
  SYNTAX_FONT_FACE_MIXED,
  SYNTAX_FONT_FACE_MULTIPLE,
  SYNTAX_GLOBAL,
  SYNTAX_KEYFRAMES,
  SYNTAX_MEDIA_QUERY,
  SYNTAX_MULTI_SELECTOR,
  SYNTAX_PROPERTIES,
  SYNTAX_PSEUDO,
  SYNTAX_MEDIA_QUERY_NESTED,
  SYNTAX_KEYFRAMES_INLINE,
  KEYFRAME_SLIDE_PERCENT,
} from '../../../tests/mocks';
import { cleanStyles } from '../../../tests/helpers';

describe('AphroditeAesthetic', () => {
  let instance: AphroditeAesthetic<any>;

  function renderAndTest(
    styles: any,
    expStyles: any,
    expClassName: string,
    global: boolean = false,
  ) {
    const nativeSheet = global
      ? instance.syntax.convertGlobalSheet(styles).toObject()
      : instance.syntax.convertStyleSheet(styles).toObject();
    const styleSheet = instance.processStyleSheet(nativeSheet);

    expect(nativeSheet).toEqual(expStyles);

    expect(instance.transformStyles(...Object.values(styleSheet))).toBe(expClassName);

    expect(cleanStyles(StyleSheetTestUtils.getBufferedStyles().join(''))).toMatchSnapshot();
  }

  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
    instance = new AphroditeAesthetic();
  });

  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('converts and transforms inline styles', () => {
    expect(instance.transformToClassName([{ margin: 0 }, { padding: 2 }])).toBe(
      'inline-0_16pg94n-o_O-inline-1_igcoje',
    );
  });

  describe('global sheet', () => {
    it('handles globals', () => {
      renderAndTest(
        SYNTAX_GLOBAL,
        {
          globals: {
            '*body': { margin: 0 },
            '*html': { height: '100%' },
            '*a': {
              color: 'red',
              ':hover': {
                color: 'darkred',
              },
            },
          },
        },
        'globals_vyedjg',
        true,
      );
    });

    it('handles @font-face', () => {
      instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE);

      expect(instance.fontFaces).toEqual({
        Roboto: [
          {
            ...FONT_ROBOTO_FLAT_SRC,
            fontFamily: ['Roboto'],
          },
        ],
      });
    });

    it('handles mixed @font-face', () => {
      instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MIXED);

      expect(instance.fontFaces).toEqual({
        Roboto: [
          {
            ...FONT_ROBOTO_FLAT_SRC,
            fontFamily: ['Roboto'],
          },
        ],
        Circular: FONT_CIRCULAR_MULTIPLE_FLAT_SRC.map(face => ({
          ...face,
          fontFamily: ['Circular'],
        })),
      });
    });

    it('handles multiple @font-face', () => {
      instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MULTIPLE);

      expect(instance.fontFaces).toEqual({
        Circular: FONT_CIRCULAR_MULTIPLE_FLAT_SRC.map(face => ({
          ...face,
          fontFamily: ['Circular'],
        })),
      });
    });

    it('handles @keyframes', () => {
      instance.syntax.convertGlobalSheet(SYNTAX_KEYFRAMES);

      expect(instance.keyframes).toEqual({
        fade: KEYFRAME_FADE,
      });
    });
  });

  describe('style sheet', () => {
    it('converts unified syntax to native syntax and transforms to a class name', () => {
      instance.fontFaces.Roboto = [FONT_ROBOTO_FLAT_SRC as any];
      instance.keyframes.fade = KEYFRAME_FADE;

      renderAndTest(
        SYNTAX_UNIFIED_LOCAL_FULL,
        {
          button: {
            margin: 0,
            padding: '6px 12px',
            border: '1px solid #2e6da4',
            borderRadius: 4,
            display: 'inline-block',
            cursor: 'pointer',
            fontFamily: [FONT_ROBOTO_FLAT_SRC],
            fontWeight: 'normal',
            lineHeight: 'normal',
            whiteSpace: 'nowrap',
            textDecoration: 'none',
            textAlign: 'center',
            backgroundColor: '#337ab7',
            verticalAlign: 'middle',
            color: 'rgba(0, 0, 0, 0)',
            animationName: [KEYFRAME_FADE],
            animationDuration: '.3s',
            ':hover': {
              backgroundColor: '#286090',
              borderColor: '#204d74',
            },
            '::before': {
              content: '"â˜…"',
              display: 'inline-block',
              verticalAlign: 'middle',
              marginRight: 5,
            },
            '@media (max-width: 600px)': {
              padding: '4px 8px',
            },
          },
        },
        'button_jr9ve',
      );
    });

    it('handles properties', () => {
      renderAndTest(SYNTAX_PROPERTIES, SYNTAX_PROPERTIES, 'props_1pbzc6n');
    });

    it('handles attribute selectors', () => {
      renderAndTest(SYNTAX_ATTRIBUTE, SYNTAX_ATTRIBUTE, 'attr_424cv8');
    });

    it('handles descendant selectors', () => {
      renderAndTest(
        SYNTAX_DESCENDANT,
        {
          list: {
            margin: 0,
            padding: 0,
            '> li': {
              listStyle: 'bullet',
            },
          },
        },
        'list_1lo5lhe',
      );
    });

    it('handles pseudo selectors', () => {
      renderAndTest(SYNTAX_PSEUDO, SYNTAX_PSEUDO, 'pseudo_q2zd6k');
    });

    it('handles multiple selectors (comma separated)', () => {
      renderAndTest(
        SYNTAX_MULTI_SELECTOR,
        {
          multi: {
            cursor: 'pointer',
            ':disabled': { cursor: 'default' },
            '[disabled]': { cursor: 'default' },
            '> span': { cursor: 'default' },
          },
        },
        'multi_1xnvdcd',
      );
    });

    it('handles inline @keyframes', () => {
      renderAndTest(
        SYNTAX_KEYFRAMES_INLINE,
        {
          single: {
            animationName: [KEYFRAME_SLIDE_PERCENT],
          },
          multiple: {
            animationName: [KEYFRAME_SLIDE_PERCENT, 'unknown', KEYFRAME_FADE],
          },
        },
        'single_1j39j3w-o_O-multiple_19eoumq',
      );
    });

    it('handles @media', () => {
      renderAndTest(
        SYNTAX_MEDIA_QUERY,
        {
          media: {
            color: 'red',
            '@media (max-width: 1000px)': {
              color: 'green',
            },
            '@media (min-width: 300px)': {
              color: 'blue',
            },
          },
        },
        'media_1yqe7pa',
      );
    });

    it('handles nested @media', () => {
      renderAndTest(
        SYNTAX_MEDIA_QUERY_NESTED,
        {
          media: {
            color: 'red',
            '@media (min-width: 300px)': {
              color: 'blue',
              '@media (max-width: 1000px)': {
                color: 'green',
              },
            },
          },
        },
        'media_y3eou6',
      );
    });
  });
});
