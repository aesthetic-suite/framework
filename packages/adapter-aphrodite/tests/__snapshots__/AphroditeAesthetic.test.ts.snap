// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AphroditeAesthetic global sheet handles globals 1`] = `"body{margin:0px !important;}html{height:100% !important;}a{color:red !important;}a:hover{color:darkred !important;}"`;

exports[`AphroditeAesthetic style sheet converts unified syntax to native syntax and transforms to a class name 1`] = `"@font-face{font-family:Roboto;font-style:normal;font-weight:normal;src:local('Robo'), url('fonts/Roboto.woff2') format('woff2'), url('fonts/Roboto.ttf') format('truetype');}@keyframes keyframe_w18mkj{from{opacity:0;}to{opacity:1;}}.button_jr9ve{margin:0px !important;padding:6px 12px !important;border:1px solid #2e6da4 !important;border-radius:4px !important;display:inline-block !important;cursor:pointer !important;font-family:\\"Roboto\\" !important;font-weight:normal !important;line-height:normal !important;white-space:nowrap !important;text-decoration:none !important;text-align:center !important;background-color:#337ab7 !important;vertical-align:middle !important;color:rgba(0, 0, 0, 0) !important;-webkit-animation-name:keyframe_w18mkj !important;animation-name:keyframe_w18mkj !important;-webkit-animation-duration:.3s !important;animation-duration:.3s !important;}.button_jr9ve:hover{background-color:#286090 !important;border-color:#204d74 !important;}.button_jr9ve::before{content:\\"â˜…\\" !important;display:inline-block !important;vertical-align:middle !important;margin-right:5px !important;}@media (max-width: 600px){.button_jr9ve{padding:4px 8px !important;}}"`;

exports[`AphroditeAesthetic style sheet handles @media 1`] = `".media_1yqe7pa{color:red !important;}@media (min-width: 300px){.media_1yqe7pa{color:blue !important;}}@media (max-width: 1000px){.media_1yqe7pa{color:green !important;}}"`;

exports[`AphroditeAesthetic style sheet handles attribute selectors 1`] = `".attr_424cv8{display:block !important;}.attr_424cv8[disabled]{opacity:0.5 !important;}"`;

exports[`AphroditeAesthetic style sheet handles descendant selectors 1`] = `".list_1lo5lhe{margin:0px !important;padding:0px !important;}.list_1lo5lhe > li{list-style:bullet !important;}"`;

exports[`AphroditeAesthetic style sheet handles inline @font-face 1`] = `"@font-face{font-family:Circular;font-style:normal;font-weight:normal;src:url('fonts/Circular.woff2') format('woff2');}@font-face{font-family:Circular;font-style:italic;font-weight:normal;src:url('fonts/Circular-Italic.woff2') format('woff2');}@font-face{font-family:Circular;font-style:normal;font-weight:300;src:url('fonts/Circular-Light.woff2') format('woff2');}@font-face{font-family:Circular;font-style:normal;font-weight:700;src:url('fonts/Circular-Bold.woff2') format('woff2');}@font-face{font-family:Roboto;font-style:normal;font-weight:normal;src:local('Robo'), url('fonts/Roboto.woff2') format('woff2'), url('fonts/Roboto.ttf') format('truetype');}.single_vfwy7z-o_O-multiple_17z04zp{font-family:\\"Circular\\",OtherFont,\\"Roboto\\" !important;}"`;

exports[`AphroditeAesthetic style sheet handles inline @keyframes 1`] = `"@keyframes keyframe_khqws7{0%{opacity:0;}50%{opacity:0.5;}100%{opacity:1;}}@keyframes keyframe_w18mkj{from{opacity:0;}to{opacity:1;}}.single_1j39j3w-o_O-multiple_19eoumq{-webkit-animation-name:keyframe_khqws7,unknown,keyframe_w18mkj !important;animation-name:keyframe_khqws7,unknown,keyframe_w18mkj !important;}"`;

exports[`AphroditeAesthetic style sheet handles multiple selectors (comma separated) 1`] = `".multi_1xnvdcd{cursor:pointer !important;}.multi_1xnvdcd:disabled{cursor:default !important;}.multi_1xnvdcd[disabled]{cursor:default !important;}.multi_1xnvdcd > span{cursor:default !important;}"`;

exports[`AphroditeAesthetic style sheet handles nested @media 1`] = `".media_y3eou6{color:red !important;}@media (min-width: 300px){.media_y3eou6{color:blue !important;}@media (max-width: 1000px){.media_y3eou6{color:green !important;}}}"`;

exports[`AphroditeAesthetic style sheet handles properties 1`] = `".props_1pbzc6n{color:black !important;display:inline !important;margin:10px !important;}"`;

exports[`AphroditeAesthetic style sheet handles pseudo selectors 1`] = `".pseudo_q2zd6k{position:fixed !important;}.pseudo_q2zd6k:hover{position:static !important;}.pseudo_q2zd6k::before{position:absolute !important;}"`;
